engine/flink/management/sample/src/main/scala/pl/touk/nussknacker/engine/management/sample/DevProcessConfigCreator.scala

```diff
@@ -76,15 +77,15 @@ class DevProcessConfigCreator extends ProcessConfigCreator {
       "monitor" -> categories(SinkFactory.noParam(EmptySink)),
       "communicationSink" -> categories(DynamicParametersSink),
       "kafka-string" -> all(new KafkaSinkFactory(new SimpleSerializationSchema[AnyRef](_, String.valueOf), processObjectDependencies) with FlinkKafkaSinkFactory),
-      "kafka-avro" -> all(new KafkaAvroSinkFactoryWithEditor(schemaRegistryProvider, processObjectDependencies))
+      "kafka-avro" -> all(new BaseKafkaAvroSinkFactoryWithEditor(schemaRegistryProvider, processObjectDependencies) with FlinkKafkaAvroSinkFactory)
```

Comment:
We have it (BaseKafka... with FlinkKafkaAvro...) in a few places. Can't we have a class (non-anonymous) for that?

Reply:
IMO current solution causes that engine specific implementation is clearly separated. Developer of next engine-specific component using similar solution won't use engine specific things in places where it is not necessary. It is very close to idea to keep definition separated from imlementation. When we add  BaseFlinkKafkaAvroSinkFactoryWithEditor it will cause that it will be a bag for "everything". I'd rather want to do the same separation in sources - but haven't done it because we have `timestampAssigner` in constructor.

Reply:
ok, when I looked further I realised I just want composition - which we discuss in other place ;)